Sources/NIO/ByteBuffer-core.swift

```diff
@@ -492,9 +492,21 @@ public struct ByteBuffer {
         return try body(.init(rebasing: self._slicedStorageBuffer.dropFirst(self.writerIndex)))
     }
 
+    /// This vends a pointer of the `ByteBuffer` at the `writeIndex`. If `minimumWritableBytes` is specified, then the `ByteBuffer` will reserve the necessary
+    /// capacity before vending the pointer to `body`.
+    ///
+    /// - warning: Do not escape the pointer from the closure for later use.
+    ///
+    /// - parameters:
+    ///     - minimumWritableBytes: The number of writable bytes to reserve capacity for before vending the `ByteBuffer` pointer to `body`. (optional parameter)
+    ///     - body: The closure that will accept the yielded bytes and return the number of bytes written.
+    /// - returns: The number of bytes written.
     @discardableResult
     @inlinable
-    public mutating func writeWithUnsafeMutableBytes(_ body: (UnsafeMutableRawBufferPointer) throws -> Int) rethrows -> Int {
+    public mutating func writeWithUnsafeMutableBytes(minimumWritableBytes: Int? = nil, _ body: (UnsafeMutableRawBufferPointer) throws -> Int) rethrows -> Int {
+        if let minimumWritableBytes = minimumWritableBytes {
+            reserveCapacity(writerIndex + minimumWritableBytes)
```

Comment:
Nit: we use explicit `self` in the NIO project, at least where we remember to.

Reply:
Ok, I'll make the change. I'm curious to hear the reasoning behind that style if you're willing to share. :)

Reply:
The reasoning is fairly straightforward: it avoids ambiguity. There is no question about where `reserveCapacity` comes from or what its side effects may be if you explicitly see that `self` is being passed to it.